libass/ass_parse.c

```diff
@@ -1000,8 +1000,8 @@ void process_karaoke_effects(ASS_Renderer *render_priv)
         switch (effect_type) {
         case EF_KARAOKE:
         case EF_KARAOKE_KO:
-            x = tm_current < tm_start ? x_start : x_end + 1;
-            break;
+            tm_end = tm_start;
+            //-fallthrough
```

Comment:
I think that instead of fallthrough it's better to move \kf branch after the switch.

Reply:
I’ve now removed the `switch` entirely and replaced the `if (k/ko) A; else if (kf) B; else error;` cascade by `if (!kf) A; B`.